docs/docs/Tutorials/ROS2/PythonTemplate.md

```diff
@@ -0,0 +1,323 @@
+# Quick Start &mdash; Writing External Controller With Github Template Repository
+---
+
+There are two github
+[template repositories](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template)
+set up (cpp/python) for a quick start on writing a
+custom controller utilizing
+[buoy_api_cpp](https://github.com/osrf/buoy_msgs/tree/main/buoy_api_cpp) and
+[buoy_api_py](https://github.com/osrf/buoy_msgs/tree/main/buoy_api_py). Please see
+[cpp examples](https://github.com/osrf/buoy_msgs/tree/main/buoy_api_cpp/examples) and
+[python examples](https://github.com/osrf/buoy_msgs/tree/main/buoy_api_py/examples) for example
+controller implementations.
+
+* [mbari_wec_template_cpp](https://github.com/mbari-org/mbari_wec_template_cpp)
+* [mbari_wec_template_py](https://github.com/mbari-org/mbari_wec_template_py)
+
+---
+
+## Using Python Template
+
+### Creating your own repo from the template
+
+You may also refer to Github's
+[template documentation](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template)
+
+To start using the python Github template
+
+1. Navigate to
+[mbari_wec_template_py](https://github.com/mbari-org/mbari_wec_template_py) and click the green
+button with the text `Use this template` and select `Create a new repository`
+
+    ![use_this_template](resources/use_this_template_py.png)
+
+2. Next, set up the repository like you would any new Github repository choosing the owner,
+repository name, public/private, etc.
+
+3. Now that your new repository is set up, clone it to your local machine, make a branch, etc.
+
+`$ git clone https://github.com/<owner>/<repo_name>.git`
+
+You should now have a python ROS2 package with the following structure:
+
+```
+<repo_name>
+    ├── config
+    │   └── controller.yaml
+    ├── launch
+    │   └── controller.launch.py
+    ├── LICENSE
+    ├── mbari_wec_template_py
+    │   ├── controller.py
+    │   └── __init__.py
+    ├── package.xml
+    ├── README.md
+    ├── resource
+    │   └── mbari_wec_template_py
+    ├── setup.cfg
+    ├── setup.py
+    └── test
+        ├── test_copyright.py
+        ├── test_flake8.py
+        └── test_pep257.py
+```
+
+### Customizing the controller
+
+You may also refer to the `README.md` in your newly cloned repository.
+
+#### Modify template for your package
+Replace `mbari_wec_template_py` with your package name and modify other fields as necessary in:
+
+- package.xml (lines 4-8)
+
+``` xml linenums="1" hl_lines="4-8" title="package.xml"
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>repo_name</name>
+  <version>3.14</version>
+  <description>Your Controller Description</description>
+  <maintainer email="your@email">Your Name</maintainer>
+  <license>Your License</license>
+```
+
+
+- setup.py (lines 7, 11, 22-25, 29)
```

Comment:
These line numbers could change if the files are changed in the template, right? Then the tutorial will have to be updated as well, not just one number, but all numbers when any line is added? Is there a way to make this less specific and more future-proof, so that people don't have to keep remembering to update in both places?

The other consideration is that, since the code block below doesn't display line numbers, it's hard for people to tell what these line numbers are talking about. It could be more clear to instead have in-line comments in the code saying, change this line, it does X, so that people understand what the code is doing.

Same for all code blocks on this page with line number references.

Reply:
It shows line numbers for me... are there really no line numbers showing up?

I tried highlighting lines but for some reason the `hl_lines` instruction in the code block isn't actually highlighting... so I just put line numbers. Also, since markdown doesn't allow referencing an actual file like .rst, this tutorial would need to be updated with any code changes anyways... But maybe the compromise is to have inline comments like you suggest

Reply:
added inline comments in code blocks. Need to confirm that others can see the line numbers in the code blocks